readthedocs/doc_builder/backends/mkdocs.py

```diff
@@ -87,7 +88,12 @@ def load_yaml_config(self):
         :raises: ``MkDocsYAMLParseError`` if failed due to syntax errors.
         """
         try:
-            return yaml.safe_load(open(self.yaml_file, 'r'),)
+            yaml_config = yaml.safe_load(open(self.yaml_file, 'r'),)
+            if not isinstance(yaml_config, dict): 
+                raise ParseError('Expected dict') 
+            if not yaml_config: 
+                raise ParseError('Unable to load YAML config') 
```

Comment:
This error is used only for the rtd  configuration file, not for the mkdocs.yml file. You should use https://github.com/stsewd/readthedocs.org/blob/59a25772e33c0c510ddb08bd01145c33675f5b62/readthedocs/doc_builder/exceptions.py#L60-L60

Also, we need a more descriptive message for the user. And a test.

Reply:
I updated the code using the MkDocsYAMLParseError, gonna work on adding the test. What would you suggest would be a good descriptive message for the user?

Reply:
@stsewd @rshrc May I suggest adding the Exception error message here:
https://github.com/stsewd/readthedocs.org/blob/59a25772e33c0c510ddb08bd01145c33675f5b62/readthedocs/doc_builder/exceptions.py#L60

and the error message could be like:
> Your MkDocs YAML configuration was incorrect. Please follow the user guide https://www.mkdocs.org/user-guide/configuration/ to configure the file properly.

If empty:

> Please make sure the MkDocs YAML configuration file is not empty.

Just my thoughts :)